server/utils/placementRequests/placementApplicationSubmissionData.ts

```diff
@@ -49,14 +52,49 @@ export const durationAndArrivalDateFromRotlPlacementApplication = (dateOfPlaceme
   }
 }
 
+export const retreivePlacementDatesFromRotlPlacementApplication = (
+  placementApplication: PlacementApplication,
+): Array<DateOfPlacement> => {
+  const datesOfPlacement = retrieveOptionalQuestionResponseFromFormArtifact(
+    placementApplication,
+    DatesOfPlacement,
+    'datesOfPlacement',
+  )
+
+  if (datesOfPlacement) {
+    return datesOfPlacement
+  }
+
+  const dateOfPlacement = {}
```

Comment:
I think this would be better as a `let` without an initializer and then we can avoid casting on line 87

Reply:
Ah, this didn't work 😢

Reply:
```   
let dateOfPlacement: DateOfPlacement

  const legacyProperties = [
    'arrivalDate',
    'arrivalDate-day',
    'arrivalDate-month',
    'arrivalDate-year',
    'duration',
    'durationDays',
    'durationWeeks',
  ]

  legacyProperties.forEach(property => {
    dateOfPlacement[property] = retrieveQuestionResponseFromFormArtifact(
      placementApplication,
      DatesOfPlacement,
      property,
    )
  })

  return [dateOfPlacement] 
  ``` 
  works for me, but not a big deal if we have to cast